velox/functions/prestosql/window/LeadLag.cpp

```diff
@@ -152,6 +152,11 @@ class LeadLagFunction : public exec::WindowFunction {
       return;
     }
 
+    // If the offset is 0 then it means always return the current row.
+    if (constantOffsetValue == 0) {
```

Comment:
We could add a member variable in the class for isConstantOffsetZero_ (like isConstantOffsetNull_). That would mean we check it only once. The condition to set rowNumbers_ can also be moved at line 147 (after isConstantOffsetNull_ case_).